pandas/core/generic.py

```diff
@@ -5092,6 +5099,47 @@ def bfill(self, axis=None, inplace=False, limit=None, downcast=None):
 
         This raises a ``TypeError`` because one of the ``dict`` keys is not of
         the correct type for replacement.
+
+        Compare the behavior of
+        ``s.replace('a', None)`` and ``s.replace({'a': None})`` to understand
+        the pecularities of the ``to_replace`` parameter.
+        ``s.replace('a', None)`` is actually equivalent to
+        ``s.replace(to_replace='a', value=None, method='pad')``,
+        because when ``value=None`` and ``to_replace`` is a scalar, list or
```

Comment:
This is interesting as I was not aware of this behavior. Certainly great to have it documented, though I would move the majority of the writing into the Notes section and shorten the blurb introducing the comparison here.